src/config.js

```diff
@@ -23,18 +25,24 @@ export function applyConfigToArgv({
   for (const option in configObject) {
     // we assume the value was set on the CLI if the default value is
     // not the same as that on the argv object as there is a very rare chance
-    // this happening
+    // of this happening
+    if (camelCase(option) !== option) {
+      throw new UsageError(`Please use camel case to specify ${option} ` +
```

Comment:
since you have the function already you can just tell them what to do (which will help them fix it faster):
````js
throw new UsageError(
  `The config option "${option}" must be specified in camel case: "${camelCase(option)}"`
);
````